src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -342,6 +342,10 @@ public function has($id)
      */
     public function get($id)
     {
+        if ('request' === $id) {
+            trigger_error('The "request" service is deprecated and will be removed in 3.0. Add a typehint for Symfony\\Component\\HttpFoundation\\Request to your controller parameters to retrieve the request instead.', E_USER_DEPRECATED);
+        }
+        
```

Comment:
I guess we have to do the same for other deprecated services such as `security.context`.